### PR TITLE
[refactor] make m_pPlayer private and expose it through method GetAppPlayer()

### PR DESCRIPTION
**Pull request by xhaggi**
see commits

### CODE REVIEW

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -4864,6 +4865,11 @@ std::string CApplication::GetCurrentPlayer()
   return m_pPlayer->GetCurrentPlayer();
 }
 
+std::shared_ptr<CApplicationPlayer> CApplication::GetAppPlayer()
+{
+  return m_pPlayer;
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: m_pPlayer is a unique_ptr. not sure if passing it away as shared_ptr make sense.

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -6048,36 +6048,36 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
   case PLAYER_CHAPTERCOUNT:
-    if(g_application.m_pPlayer->IsPlaying())
-      strLabel = StringUtils::Format("%02d", g_application.m_pPlayer->GetChapterCount());
+    if(g_application.GetAppPlayer()->IsPlaying())
+      strLabel = StringUtils::Format("%02d", g_application.GetAppPlayer()->GetChapterCount());
     break;
   case PLAYER_CHAPTERNAME:
-    if(g_application.m_pPlayer->IsPlaying())
-      g_application.m_pPlayer->GetChapterName(strLabel);
+    if(g_application.GetAppPlayer()->IsPlaying())
```
Comment (FernetMenta) [inline xbmc/GUIInfoManager.cpp]: with regard on performance we should think about a better solution. now that's 2 function calls extra and 2 times a shared ptr get constructed/destructed.

File: xbmc/GUIInfoManager.cpp [hunk 3]
```diff
@@ -6274,22 +6274,22 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
   case VIDEOPLAYER_STEREOSCOPIC_MODE:
-    if(g_application.m_pPlayer->IsPlaying())
+    if(g_application.GetAppPlayer()->IsPlaying())
     {
       strLabel = m_videoInfo.stereoMode;
     }
     break;
   case VIDEOPLAYER_SUBTITLES_LANG:
-    if(g_application.m_pPlayer && g_application.m_pPlayer->IsPlaying() && g_application.m_pPlayer->GetSubtitleVisible())
+    if(g_application.GetAppPlayer() && g_application.GetAppPlayer()->IsPlaying() && g_application.GetAppPlayer()->GetSubtitleVisible())
```
Comment (FernetMenta) [inline xbmc/GUIInfoManager.cpp]: here it is even worse. you could get a ref to player and then use this for the following calls.

File: xbmc/Application.h [hunk 4]
```diff
@@ -502,6 +501,7 @@ class CApplication : public CXBApplicationEx, public IPlayerCallback, public IMs
   std::atomic_uint m_WaitingExternalCalls;        /*!< counts threads wich are waiting to be processed in FrameMove */
   unsigned int m_ProcessedExternalCalls;          /*!< counts calls wich are processed during one "door open" cycle in FrameMove */
   unsigned int m_ProcessedExternalDecay = 0;      /*!< counts to close door after a few frames of no python activity */
+  std::shared_ptr<CApplicationPlayer> m_pPlayer;
```
Comment (FernetMenta) [inline xbmc/Application.h]: now thinking more about it I'd say that a pointer makes no sense here. It never gets deleted and nothing checks the pointer for being valid. Better make it a normal member. CAppPlayer wraps the actual player and does the checks for nullptr.

Comment (xhaggi) [reply xbmc/Application.h]: Ok will change it to be a normal member like we did for CSeekHandler

Comment (xhaggi) [reply xbmc/Application.h]: done + force-pushed

File: xbmc/GUIInfoManager.cpp [hunk 5]
```diff
@@ -6136,11 +6136,11 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
     }
     break;
   case PLAYER_PLAYSPEED:
-      if(g_application.m_pPlayer->IsPlaying())
+      if(g_application.GetAppPlayer().IsPlaying())
```
Comment (FernetMenta) [inline xbmc/GUIInfoManager.cpp]: all those condition checks are rather useless/wrong and can be dropped
- there is no guarantee that player is still playing when entering the block
- if player is not playing the methods return defaults
- unlikely that a skins query those labels if player is not playing

File: xbmc/GUIInfoManager.cpp [hunk 6]
```diff
@@ -6225,38 +6225,38 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
   case VIDEOPLAYER_VIDEO_ASPECT:
-    if (g_application.m_pPlayer->IsPlaying())
+    if (g_application.GetAppPlayer().IsPlaying())
     {
       strLabel = CStreamDetails::VideoAspectToAspectDescription(CServiceBroker::GetDataCacheCore().GetVideoDAR());
     }
     break;
   case VIDEOPLAYER_AUDIO_CHANNELS:
-    if(g_application.m_pPlayer->IsPlaying())
+    if(g_application.GetAppPlayer().IsPlaying())
```
Comment (FernetMenta) [inline xbmc/GUIInfoManager.cpp]: also not needed. m_audio/videoInfo can be reset to defaults, if player is not playing

File: xbmc/SeekHandler.cpp [hunk 7]
```diff
@@ -202,9 +202,9 @@ int CSeekHandler::GetSeekSize() const
 
 void CSeekHandler::SetSeekSize(double seekSize)
 {
-  int64_t playTime = g_application.m_pPlayer->GetTime();
-  double minSeekSize = (g_application.m_pPlayer->GetMinTime() - playTime) / 1000;
-  double maxSeekSize = (g_application.m_pPlayer->GetMaxTime() - playTime) / 1000;
+  int64_t playTime = g_application.GetAppPlayer().GetTime();
```
Comment (FernetMenta) [inline xbmc/SeekHandler.cpp]: better requesting the ref only once

File: xbmc/GUIInfoManager.cpp (hunk 6 above)
Comment (xhaggi) [reply xbmc/GUIInfoManager.cpp]: done

File: xbmc/GUIInfoManager.cpp (hunk 5 above)
Comment (xhaggi) [reply xbmc/GUIInfoManager.cpp]: done in a separate commit

**Comment by xhaggi** [comment]
@Voyager1 FYI .. I changed the unique ptr `CApplicationStackHelper` to be a plain member of `CApplication` and made it private with a public method `GetAppStackHelper`.

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by xhaggi** [comment]
@FernetMenta would be nice if you could take another look so I don't need another rebase round.

File: xbmc/GUIInfoManager.cpp [hunk 8]
```diff
@@ -7223,88 +7190,88 @@ bool CGUIInfoManager::GetBool(int condition1, int contextWindow, const CGUIListI
     CGUIWindowSlideShow *slideShow = g_windowManager.GetWindow<CGUIWindowSlideShow>(WINDOW_SLIDESHOW);
     bReturn = (slideShow && slideShow->GetCurrentSlide() && slideShow->GetCurrentSlide()->IsVideo());
   }
-  else if (g_application.m_pPlayer->IsPlaying())
+  else if (g_application.GetAppPlayer().IsPlaying())
```
Comment (FernetMenta) [inline xbmc/GUIInfoManager.cpp]: why did you keep this?

Comment (xhaggi) [reply xbmc/GUIInfoManager.cpp]: I'm not sure if that is fine to remove all `IsPlaying` check in `CGUIInfoManager `

Comment (xhaggi) [reply xbmc/GUIInfoManager.cpp]: dropped all `IsPlaying` checks from GUIInfoManager .. runtime tested and it looks good.

**Comment by FernetMenta** [comment]
+1

**Comment by Voyager1** [comment]
> I changed the unique ptr CApplicationStackHelper to be a plain member of CApplication and made it private with a public method GetAppStackHelper.

seems ok to me.

**Comment by xhaggi** [comment]
Jenkins build this please

**Pull request merged**